packages/react-tooltip/Spec.md

```diff
@@ -1,63 +1,474 @@
-# @fluentui/react-tooltip Spec
+# Tooltip Spec
 
-## Background
-
-_Description and use cases of this component_
+Tooltips provide additional information about an element when hovering or focusing on the element.
 
 ## Prior Art
 
-_Include background research done for this component_
+- OpenUI Tooltip resarch: https://open-ui.org/components/tooltip.research
+- GitHub Epic issue: [Tooltip Convergence #16735](https://github.com/microsoft/fluentui/issues/16735)
+
+### Tooltips in v8/Fabric
+
+v8 tooltips use a `TooltipHost` wrapped around the target element to provide tooltip functionality. This creates a `div` around the element that listens for mouse and focus events.
+
+The `Tooltip` component renders as a `Callout`, and supports all `Callout` props.
+
+```jsx
+<TooltipHost
+  content="This is the tooltip content"
+  // This id is used on the tooltip itself, not the host
+  // (so an element with this id only exists when the tooltip is shown)
+  id={tooltipId}
+  calloutProps={calloutProps}
+  styles={hostStyles}
+>
+  <DefaultButton aria-describedby={tooltipId}>Hover over me</DefaultButton>
+</TooltipHost>
+```
+
+#### Drawbacks
+
+There are a few drawbacks with this approach to adding tooltips, which are outlined in [☂ Tooltip: open issues to resolve in converged approach #15102](https://github.com/microsoft/fluentui/issues/15102), and summarized below:
+
+- The wrapper `div` created by `TooltipHost` can cause layout issues for the component. It also doesn't always result in proper positioning for the tooltip.
+- The API is overly complex.
+- It's not possible to have a tooltip within a tooltip
+- There's no coordination between tooltips on a page. For example, moving the mouse between two elements with tooltips should cause the second tooltip to appear immediately without fading in/out.
+
+### Tooltips in v0/Northstar
+
+v0 tooltips use a `trigger` property to render the tooltip's target component. However, unlike v8 it does not create a wrapper `div` around the target component, but instead adds listeners to the target component's props.
+
+```jsx
+<Tooltip content="Example tooltip" trigger={<Button content="A button" />} />
+```
+
+# Sample Code
+
+There are a few ways to add a tooltip to a component.
+
+## A component with a `tooltip` slot
```

Comment:
Having a tooltip slot will make things very clean. I do have one concern about what components should have tooltip slots and where ? 

We should consider the risk be for API consistency in the future and how we handle requests like 

> Can you add a tooltip slot to XYZ component cause our team needs it

😄

Reply:
The answer to that request is "good news, you can add it yourself!". As long as the component is built following our state hooks pattern, then all they need to do is add the `useTooltipSlot` hook.

For example, I've been testing a prototype implementation by creating an `AvatarWithTooltip` component:
```tsx
const AvatarWithTooltip = React.forwardRef((props: AvatarWithTooltipProps, ref: React.Ref<HTMLElement>) => {
  const state = useAvatar(props, ref, { tabIndex: 0 });
  useTooltipSlot(state);
  useAvatarStyles(state);
  return renderAvatar(state);
});
```

Reply:
That said, we'll need to make sure we add the `tooltip` prop to all of our core components that should support them.

Reply:
I suppose that wouldn't work well.. only one of them would show up but I'm not sure offhand which one. That said, it's equally as easy to write the wrong thing like this:
```tsx
<TooltipTrigger tooltip="from outer trigger">
  <TooltipTrigger tooltip="from inner trigger">
    <Button />
  </TooltipTrigger>
</TooltipTrigger>
```

It may be worthwhile to specify _which_ one shows up, since we can't really prevent it. (Probably the outermost one should take priority?)

I really like the idea of the `tooltip="..."` prop, which is effectively syntactic sugar for wrapping with a `<TooltipTrigger>`, but is much cleaner and more straightforward to understand.

Reply:
Imo, `tooltip` is easier but more confusing. We're much more verbose with the component instead of the prop and we're also consistent with the rest of our library.
Also, we avoid confusion of what is part of the component itself or an addition, as well as not needing the cross dependency.

Reply:
We had a poll in the spec review meeting, and had a strong majority of people preferring having only `TooltipTrigger`, and not an additional `tooltip` prop.
* (6) - Only `TooltipTrigger`
* (1) - Add `tooltip` prop
* (2) - No vote